index.js

```diff
@@ -1,12 +1,19 @@
 'use strict';
-module.exports = (arr, comp) => {
-	for (let i = 0; i < arr.length - 1; i++) {
-		const a = arr[i];
-		const b = arr[i + 1];
 
-		if (comp ? comp(a, b) > 0 : a > b) {
+const defaultComparator = (a, b) => a > b ? 1 : 0;
+module.exports = (arr, comp) => {
+	// pass only iterable objects
+	if (!arr || typeof arr[Symbol.iterator] !== 'function') {
+		return false;
+	}
+	comp = comp || defaultComparator;
+	var a = arr[0];
+	for (var i = 1, len = arr.length; i < len; i++) {
```

Comment:
Nope, native c++ code will be slower for this case because we would get bottleneck on bridging between C++ and JS. I know it because wrote native module for V8 for deep cloning. That's why some build-in function such as ```Array.proptotype.sort``` (https://github.com/v8/v8/blob/950d2051a5ff065a5bc1d31f0e5d1bba850d0b3c/src/array.js) wrote on JS-side into private area